app/models/container_deployment.rb

```diff
@@ -0,0 +1,11 @@
+class ContainerDeployment < ApplicationRecord
+  has_many :container_deployment_authentications
+  belongs_to :deployed_ems, :class_name => 'ExtManagementSystem'
+  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem'
```

Comment:
@durandom it has only one has many because a single ems can only serve one role either deployed_ems (the openshift provider we deploy) or deployed_on_ems (the ems we use to deploy on).